File: backend/app/schemas/payment.py
# app/schemas/payment.py - Clean VNPay Payment Schemas (Zero Stripe / Demo)
from datetime import datetime
from typing import Any, Dict, List, Optional
from pydantic import BaseModel, Field


class CreatePaymentTransactionRequest(BaseModel):
    product_type: str = Field(..., description="PLAN or STORAGE_ADDON")
    product_id: int = Field(..., description="ID of the plan or storage addon")
    billing_cycle: str = Field(default="monthly", description="monthly or yearly")
    payment_method: str = Field(default="VNPAY", description="VNPAY payment gateway")


class PaymentTransactionResponse(BaseModel):
    id: int
    user_id: int
    transaction_code: str
    amount: float
    currency: str = "VND"
    payment_method: str
    status: str  # pending, completed, failed, cancelled, expired
    product_type: Optional[str] = None
    product_id: Optional[int] = None
    product_name: Optional[str] = None
    billing_cycle: Optional[str] = None
    metadata: Optional[Dict[str, Any]] = None
    created_at: datetime
    updated_at: datetime


class PaymentTransactionListResponse(BaseModel):
    transactions: List[PaymentTransactionResponse]
    total: int


class VNPayReturnResponse(BaseModel):
    status: str
    message: str
    transaction_code: Optional[str] = None
    is_active: bool = False
    is_success: Optional[bool] = None
    amount_vnd: Optional[float] = None
    transaction: Optional[PaymentTransactionResponse] = None

File: frontend/src/types/payment.ts
export type PaymentMethod = "VNPAY";
export type PaymentStatus = "pending" | "completed" | "failed" | "cancelled" | "expired";

export interface CreatePaymentTransactionRequest {
  product_type: "PLAN" | "STORAGE_ADDON";
  product_id: number;
  billing_cycle?: "monthly" | "yearly";
  payment_method?: PaymentMethod;
}

export interface PaymentTransaction {
  id: number;
  user_id: number;
  transaction_code: string;
  amount: number;
  currency: string;
  payment_method: PaymentMethod | string;
  status: PaymentStatus | string;
  product_type?: "PLAN" | "STORAGE_ADDON";
  product_id?: number;
  product_name?: string;
  billing_cycle?: "monthly" | "yearly";
  metadata?: Record<string, any>;
  created_at: string;
  updated_at: string;
}

export interface PaymentTransactionListResponse {
  transactions: PaymentTransaction[];
  total: number;
}
